mux/src/domain.rs

```diff
@@ -84,7 +84,7 @@ pub trait Domain: Downcast + Send + Sync {
             None => anyhow::bail!("Invalid tab id {}", tab),
         };
 
-        let pane_index = match tab
+        let mut pane_index = match tab
```

Comment:
with the changes suggested below, we can keep this immutable, which makes the code overall easier to reason about:

```suggestion
        let pane_index = match tab
```